src/main/groovy/com/morpheusdata/hyperv/sync/NetworkSync.groovy

```diff
@@ -0,0 +1,156 @@
+                    morpheusItem?.externalId == cloudItem?.name
+                }.onDelete { removeItems ->
+                    morpheusContext.async.cloud.network.remove(removeItems).blockingGet()
+                }.onUpdate { List<SyncTask.UpdateItem<Network, Map>> updateItems ->
+                    updateMatchedNetworks(updateItems, networkTypes, subnetType, server)
+                }.onAdd { itemsToAdd ->
+                    addMissingNetworks(itemsToAdd, networkTypes, subnetType, server)
+                }.withLoadObjectDetailsFromFinder { List<SyncTask.UpdateItemDto<NetworkIdentityProjection, Map>> updateItems ->
+                    return morpheusContext.async.cloud.network.listById(updateItems.collect { it.existingItem.id } as List<Long>)
+                }.start()
+            } else {
+                log.error("Error not getting the listNetworks")
+            }
+        } catch (e) {
+            log.error("cacheNetworks error: ${e}", e)
+        }
+    }
+
+    private addMissingNetworks(Collection<Map> addList, List<NetworkType> networkTypes, NetworkSubnetType subnetType, ComputeServer server) {
+        log.debug("NetworkSync >> addMissingNetworks >> called")
+        def networkAdds = []
+        try {
+            addList?.each { cloudItem ->
+                def networkType = networkTypes.find{ type -> type.externalType == cloudItem.type}
+                def networkConfig = [
+                        code        : "hyperv.network.${cloud.id}.${cloudItem.name}",
+                        category    : "hyperv.network.${cloud.id}",
+                        cloud        : cloud,
+                        dhcpServer  : true,
+                        name        : cloudItem.name,
+                        externalId  : cloudItem.name,
+                        type        : networkType,
+                        refType     : 'ComputeZone',
+                        refId       : cloud.id,
+                        owner       : cloud.owner,
+                        active      : cloud.defaultNetworkSyncActive
+                ]
+                Network networkAdd = new Network(networkConfig)
+
+                def subnetConfig = [
+                        account             : server.account,
+                        category            : "hyperv.network.${cloud.id}.${server.id}",
+                        networkSubnetType   : subnetType,
+                        code                : "hyperv.network.${cloud.id}.${server.id}.${cloudItem.name}",
+                        name                : cloudItem.name,
+                        externalId          : cloudItem.name,
+                        refType             : 'ComputeServer',
+                        refId               : server.id,
+                        description         : cloudItem.name
+                ]
+
+                def addSubnet = new NetworkSubnet(subnetConfig)
+                morpheusContext.async.networkSubnet.create([addSubnet], networkAdd)
```

Comment:
Has the results of adding networks with subnets been verified? I don't think this will work. At the time of subnet create `networkAdd` doesn't have an ID.  The subnets might need to be added to a subnet save list and save a networks subnets after all networks have been created.

Reply:
@ddeyoung changes has been done as suggested